### PR TITLE
Branch4

### PR DESCRIPTION
**Pull request by tranquanlam**

### CODE REVIEW

**Comment by tranquanlam** [comment]
ready

File: .gitignore [hunk 1]
```diff
@@ -4,7 +4,15 @@
...
 .pnp.js
+<<<<<<< HEAD
+<<<<<<< HEAD
 yarn.lock
+=======
+
+>>>>>>> b833af2... CartAPILocalstoge
+=======
+yarn.lock
+>>>>>>> 19b0f8c... branch3.1
```
Comment (phamvanbk11) [inline .gitignore]: (tat)

**Comment by phamvanbk11** [comment]
commented

**Comment by tranquanlam** [comment]
ready

**Comment by tranquanlam** [comment]
ready

File: src/component/ADMIN/element/chart.js [hunk 2]
```diff
@@ -54,7 +54,13 @@ class chart extends Component {
             <div className="page-wrapper chiller-theme toggled">
                 <Sidebar></Sidebar>
                 <main className="page-content">
+<<<<<<< HEAD
                     <Doughnut data={data} />
+=======
+                    <h3>Tỉ lệ số lượng nhân viên </h3>
+                    <Doughnut data={data} />
+                    <h3>Doanh Thu hàng tháng </h3>
+>>>>>>> branch4
```
Comment (phamvanbk11) [inline src/component/ADMIN/element/chart.js]: (tat)

File: src/component/ADMIN/element/editUser.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            password :this.state.txtPassword,
+            address : this.state.txtAdrress
+          }
+          this.setState({
+            isHasToPage :true
+          })
+          axios.put(`http://5d08a7b5034e5000140106c4.mockapi.io/api/user/${id}`, { ...user })
+    }
+    async componentDidMount() {
+        let listuser = await axios.get('http://5d08a7b5034e5000140106c4.mockapi.io/api/user')
```
Comment (phamvanbk11) [inline src/component/ADMIN/element/editUser.js]: Chuyển sang redux luôn

File: src/component/ADMIN/element/product.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    constructor(props) {
+        super(props)
+        this.state = {
+            title: '',
+          lispro: []
+        }
+      }
+    async componentDidMount(){
+        var list =[]
+        let data = await axios.get('http://5d08a7b5034e5000140106c4.mockapi.io/api/products')
```
Comment (phamvanbk11) [inline src/component/ADMIN/element/product.js]: Chuyển sang redux luôn

File: src/component/ADMIN/element/table.js [hunk 5]
```diff
@@ -45,17 +48,23 @@ class table extends Component {
             .then(function (response) {
                 return response.data
             })
-
+        let listdataUser = await axios.get('http://5d08a7b5034e5000140106c4.mockapi.io/api/user')
```
Comment (phamvanbk11) [inline src/component/ADMIN/element/table.js]: get data chuyển sang redux luôn

**Comment by phamvanbk11** [comment]
commented

**Comment by tranquanlam** [comment]
ready

**Pull request merged**